### PR TITLE
Add cstor apis

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <ashutosh.kumar@mayadata.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Copyright © 2017 The OpenEBS Authors
```
Comment (prateekpandey14) [inline Makefile]: nit: 2020

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module github.com/openebs/api
```
Comment (prateekpandey14) [inline go.mod:1]: lets have module name as `openebs.io/api` , just change `module openebs.io/api`

Comment (prateekpandey14) [reply go.mod:1]: Chaging the module name you have to make changes in code-gen as well, to generate with `openebs.io` import path

File: pkg/apis/cstor/v1/cstorvolume.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+package v1
+
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (prateekpandey14) [inline pkg/apis/cstor/v1/cstorvolume.go:26]: add  openapi gen tag for all the resources `// +k8s:openapi-gen=true`

File: pkg/apis/cstor/v1/cstorvolumereplica.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	// create this cvr
+	SourceVolumeKey CVRKey = "openebs.io/source-volume"
+
+	// SnapshotNameKey stores the name of the snapshot being used to restore this replica
+	SnapshotNameKey CVRKey = "openebs.io/snapshot"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (prateekpandey14) [inline pkg/apis/cstor/v1/cstorvolumereplica.go:40]: add openapi gen tag // +k8s:openapi-gen=true

File: pkg/apis/cstor/v1/cstorvolumereplica.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+// +resource:path=cstorvolumereplica
```
Comment (prateekpandey14) [inline pkg/apis/cstor/v1/cstorvolumereplica.go:41]: we don't need this tag anymore

File: go.mod (hunk 2 above)
Comment (sonasingh46) [reply go.mod:1]: As discussed -- will do this later once we migrate our operators.

File: pkg/apis/cstor/v1/cstorvolume.go (hunk 3 above)
Comment (prateekpandey14) [reply pkg/apis/cstor/v1/cstorvolume.go:26]: As discussed , we will be fixing above changes in upcmoing PR's

File: pkg/apis/cstor/v1/cstorvolumereplica.go (hunk 5 above)
Comment (prateekpandey14) [reply pkg/apis/cstor/v1/cstorvolumereplica.go:41]: As discussed , we will be fixing above changes in upcmoing PR's

**Pull request merged**